src/launch/linux.rs

```diff
@@ -83,166 +79,84 @@ impl From<i32> for TdxError {
+pub struct TdxInitMemRegion {
+    /// Host physical address of the target page to be added to the TD
+    pub source_addr: u64,
 
-    /// CPUID configuration information for the EDX register.
-    pub edx: u32,
-}
+    /// Guest physical address to be mapped
+    pub gpa: u64,
 
-/// Provides information about the Intel TDX module. This is equivalent to
-/// `struct kvm_tdx_capabilities` in the kernel.
-#[derive(Debug)]
-#[repr(C)]
-pub struct Capabilities {
-    /// Bitmap where if any certain bit is 0, it must be 0 in any TD's
-    /// ATTRIBUTES, which specifies various guest TD attributes. The value of
-    /// this field reflects the Intel TDX module capabilities and configuration
-    /// and CPU capabilities.
-    pub attrs_fixed0: u64,
-
-    /// Bitmap where if any certain bit is 1, it must be 1 in any TD's
-    /// ATTRIBUTES, which specifies various guest TD attributes. The value of
-    /// this field reflects the Intel TDX module capabilities and configuration
-    /// and CPU capabilities.
-    pub attrs_fixed1: u64,
-
-    /// Bitmap where if any certain bit is 0, it must be 0 in any TD's XFAM.
-    /// XFAM (eXtended Features Available Mask) determines the set of extended
-    /// features available for use by the guest TD.
-    pub xfam_fixed0: u64,
-
-    /// Bitmap where if any certain bit is 1, it must be 1 in any TD's XFAM.
-    /// XFAM (eXtended Features Available Mask) determines the set of extended
-    /// features available for use by the guest TD.
-    pub xfam_fixed1: u64,
-
-    /// Supported Guest Physical Address Width
-    pub supported_gpaw: u32,
-
-    /// Padding space. Ignored
-    _padding: u32,
-
-    /// Reserved space. Ignored.
-    _reserved: [u64; 251],
-
-    /// Number of CPUID_CONFIG entries
-    pub nr_cpuid_configs: u32,
-
-    /// Enumeration of the CPUID leaves/sub-leaves that contain bit fields whose
-    /// virtualization by the Intel TDX module is either:
-    ///
-    /// - Directly configurable (CONFIG_DIRECT) by the host VMM
-    /// - Bits that the host VMM may allow to be 1 (ALLOW_DIRECT) and their
-    ///   native value, as returned by the CPU, is 1
-    ///
-    /// Note that the virtualization of many CPUID bit fields not enumerated in
-    /// this list is configurable indirectly via the XFAM and ATTRIBUTES assigned
-    /// to a TD by the host VMM.
-    pub cpuid_configs: [CpuidConfig; NR_CPUID_CONFIGS],
+    /// Number of pages to be mapped
+    pub nr_pages: u64,
 }
 
-impl Default for Capabilities {
-    fn default() -> Self {
-        Self {
-            attrs_fixed0: 0,
-            attrs_fixed1: 0,
-            xfam_fixed0: 0,
-            xfam_fixed1: 0,
-            supported_gpaw: 0,
-            _padding: 0,
-            _reserved: [0; 251],
-
-            nr_cpuid_configs: NR_CPUID_CONFIGS as u32,
-            cpuid_configs: [Default::default(); NR_CPUID_CONFIGS],
-        }
-    }
-}
+#[allow(non_camel_case_types)]
+pub type __u64 = ::std::os::raw::c_ulonglong;
```

Comment:
I don't think this is necessary, you can implement the types as `u64`. In C it's necessary (due to arch constraints) but not in Rust.